### DISCUSSION TITLE
Firewalls

### DISCUSSION THREAD

**Discussion by ghost** [HotCakeX/Harden-Windows-Security#397] [Q&A]
Hi!

Do you recommend not having any 3rd party firewalls like portmaster? Or do you recommend to block things at a dns level using your module?

Thank you!

**Comment by HotCakeX** [comment]
Hi, I don't recommend using a 3rd party Firewall. Windows Firewall is already very capable and has lots of features. If you let me know what you want to do i'll be able to help better.

**Reply by ghost** [reply]
Thank you for the reply. I was thinking like being able to block ads and trackers and things like that. Portmaster or simplewall does a pretty good job at that

**Reply by HotCakeX** [reply]
No problem, those are good options because Windows Firewall works with IP addresses instead of domain names so for ad blocking they're better. Alternatively you can move the entire process off of your device and use something like NextDNS or similar service that don't require you to install anything or use your own device's processing power, or keeping the block lists up to date.

**Reply by HotCakeX** [reply]
By the way device wide ad blocking is only suitable if you have many non-browser apps that show ads. If browsers are where you see most of the ads then ublock origin is better or a browser extension in general is a better ad blocker since it can directly interact with the page content, for example to block YouTube video ads.

**Reply by ghost** [reply]
Oh that makes a lot of sense. Thank you! I really want to follow your recommendations haha! That’s why I just want to double check EVERYTHING if it’s not in your guide haha

**Reply by HotCakeX** [reply]
Oh I'm flattered, thank you! :)

**Reply by HotCakeX** [reply]
* Navigate to "Windows Defender Firewall with Advanced Security"
* Select either Inbound or Outbound Rules from the left pane

<br>

![image](https://github.com/user-attachments/assets/854797ce-44af-4f67-93aa-b55e8e0b463e)

<br>

Select "New Rule" from the right pane

<br>

![image](https://github.com/user-attachments/assets/ee460639-d047-469d-8fa5-eaf8e7b18f9f)

<br>

Then follow the wizard

<br>

![image](https://github.com/user-attachments/assets/de265b76-df00-4831-afaf-76c4740fe45b)


<br>

![image](https://github.com/user-attachments/assets/90300e59-008e-4265-8c2d-774d5956200f)


<br>

![image](https://github.com/user-attachments/assets/84d928b2-43ed-4a54-9ec7-d4cf6f19efe3)


<br>

![image](https://github.com/user-attachments/assets/782a7d2e-df08-4d70-9d9c-a6c9c23e6e88)


<br>

![image](https://github.com/user-attachments/assets/754d6b1f-67cd-4666-84a4-06d78dda26f1)


<br>

That's it. If you were doing it for Inbound rules, repeat the same steps for Outbound rules next.

<br>

**Reply by HotCakeX** [reply]
The method is right but those programs might be using different executables for different tasks, so instead of browsing for a single exe, you'd need to scan the installation folder of an app for all of the executables and add them to block lists.

I'm planning to add automation for this task and more advanced scenarios to one of the tools in this repo soon.

**Comment by HotCakeX** [comment]
Windows Firewall now supports DNS blocking, so you're no longer limited to only using IP addresses.

https://learn.microsoft.com/en-us/windows/security/operating-system-security/network-security/windows-firewall/dynamic-keywords

I will be implementing it in some form in the future
